### PR TITLE
Fix re-creating the swapchain causing a crash

### PR DESCRIPTION
**Pull request by ZNixian**
As per the commit message:

Previously, when the swapchain was re-created (which notably occurs in
Blaze4D when the main window is resized) the application would crash
with FboManager.setMainFbo complaining that a main FBO was already set.

This was caused by not properly clearing references to the old FBOs that
were already set. Clearing those out allows us to set the new FBO, but
still won't allow a new one to be set without clearing all of the FBOs.

I've had a look and can't find any reason why it would, but it's
possible that changing the main FBO could cause some other issue I
didn't notice.

Additionally, this patch avoids deleting the descriptor pools while
freeing the swapchain. They're freed by the shaders (so shouldn't cause
a resource leak on shutdown), and the shaders themselves aren't freed
when the swapchain is freed. This was previously being done incorrectly
anyway, since it left descriptor sets still attached to the now-deleted
pool - this would cause a validation failure (and on my NVidia drivers,
a crash) when freeing those descriptor sets.

I can't think of any reason why the descriptor pools would need to be
freed when deleting the swapchain (though I'm not especially familier
with Vulkan so I may be wrong about this). While I think you can have
descriptor sets that point to the FBOs or the swapchain images, those
can and should be deleted by themselves.

### CODE REVIEW

**Review by ramidzkh — APPROVED** [review]

File: src/main/java/graphics/kiln/rosella/render/renderer/Renderer.java [hunk 1]
```diff
@@ -203,14 +203,6 @@ private void recreateSwapChain(Display window, Rosella rosella) {
     }
 
     public void freeSwapChain() {
-        for (RawShaderProgram shader : common.shaderManager.getCachedShaders().keySet()) {
-            if (shader.getDescriptorPool() != VK_NULL_HANDLE) {
```
Comment (hYdos) [inline src/main/java/graphics/kiln/rosella/render/renderer/Renderer.java:207]: Is this related to this PR?

Comment (burgerindividual) [reply src/main/java/graphics/kiln/rosella/render/renderer/Renderer.java:207]: This is done in more than one place, and it's outlined in the description in the PR. This should be OK.

**Review by burgerindividual — APPROVED** [review]

**Review by hYdos — APPROVED** [review]

**Comment by hYdos** [comment]
Ramid, Get burger to check his emails.

**Comment by burgerindividual** [comment]
why

**Comment by hYdos** [comment]
do it pls

**Comment by ramidzkh** [comment]
we have git at school
git at school:

**Comment by hYdos** [comment]
conc
